### DISCUSSION TITLE
Trying to switch from OneNote, but having some issues. Help me succeed?

### DISCUSSION THREAD

**Discussion by kazeookami** [TriliumNext/Trilium#3849] [General]
So the big beauty that's making Trilium attractive for me is the ability to nest documents deeper than 2 laters, and the ability collapse the inner layers separately from the top layer. That's a huge plus for me.

There are two issues I find troublesome though, one is a mild inconvenience and the other may be the deal breaker.

The mild inconvenience is that I can't rapidly share info from Trilium to Discord. By rapidly I mean a single copy/paste action. With OneNote if I want to share an image of a building and a description of it I can just hit CTRL+A in OneNote, Hit CTRL+C, then go to Discord and hit CTRL+A and enter. They have the image and the description all together all at once. It posts it as an image, not an image then text, but it's a single action share.

The second one is more of an issue. I use tables a lot. Tables that are ever-changing. In OneNote when I want to start a table I just need a letter, a character of any kind. If I know I am going to need a 7-column table because I am building something for a 6-character D&D party I just drop a period, then I hit TAB 7 times. I now have a table that's 1 row and 7 columns. I need a new row, I just hold CTRL and hit enter and it gives me a new row. Doesn't matter if I am at the top of the table, the middle of the table, or the end of the table. I can jump into an existing table on row 17 and decide I need something new at row 18. I hit CTRL+Enter and it will drop me down to a new empty row 18 while moving the entire rest of the table down a row. The ability to do this all with the keyboard without taking my hands off and delaying is important to me.  Additionally, if a row is no longer useful to me I can just hit backspace through the empty cells and when I am in the first cell, hit backspace once more and it removes the row splitting the table in half. Hit backspace once more and it merges the two tables back together with the row gone. 

Just yesterday I made my first entire table in Trilium for a new Star Trek game I am running. 4 players, put all their stats and skills into the table. I didn't know the length of the table in the beginning and I wanted to put all the info in alphabetically but not all players have the same skills. So I had a lot of jumping up and down the table to do adding new rows in the middle sporadically. Just like I often do in OneNote. In Trilium because I have to take my hands off the keyboard, use the mouse to hover over the table, wait to see what side the toolbar is going to pop up on then click the toolbar to unfold its menu to select add a row. The buttons on the toolbar not always working the first click either. Overall it end up taking 10 times more effort and about 5 times more time to get lesser results than if I did it in OneNote. After all in OneNote I can also CTRL+A, CTRL+C, and CTRL+V to send an entire table to someone in Discord and they get the whole table with its formatting all intact.  With Trilium it tries to send it all as plain text, all formatting lost. Sure I can do screenshots with Lightshot or something else but now that table is being sent at 3 images instead of 1

In summary, the ability to nest pages deeper and collapse nests on any level makes me want to switch to Trilium. It's an awesome feature that I've spent years bitching about OneNote not having. But having such a slow clumsy and cumbersome table interface would be a bigger deal and would keep me away. My hope is, I can be shown a way to speed up table usage without the need to take my hands off the keyboard.

**Comment by XXXJetfireXXX** [comment]
Well, tables are a CKEditor thing, so I guess they're the only ones who can do something about it...
Also yeah, sharing something from Trilium to Discord may be somewhat problematic, especially when it comes to something like tables. I use Trilium mainly to just store all my thoughts, ideas, concepts and other kinds of stuff, which I don't need to share ever so often. Though, copying text works fine, which is enough to me in most cases.
Speaking of images, you can copy an image to clipboard by opening the image note itself (double click on an image reference inside a text note) and right-clicking on it:
![image](https://user-images.githubusercontent.com/43646178/232340656-601b6476-c87e-4897-8754-c4bfe420db97.png)
Not so convenient, but it works. @zadam is planning on adding a new object type - note attachments, which might help your situation with copying images since right now, all images in text notes are actually references, the images themselves are separate notes.

**Comment by meichthys** [comment]
It may not be desirable, but you can also share the whole Trilium note using the `Shared` toggle, and then paste the link into Discord:
![image](https://user-images.githubusercontent.com/10717998/236274639-9825fc9d-c151-45eb-82cc-96a749213e51.png)


You'd just want to make sure the note only contains the information you want to share.

Also, we've recently added a OneNote section to the wiki. Be sure to take a look there if you do decide to make the migration:
https://github.com/zadam/trilium/wiki/Onenote
